### PR TITLE
fix: remove duplicate error callback in storage

### PR DESCRIPTION
**Pull request by Jordan-Nelson**
*Issue #, if available:* https://github.com/aws-amplify/amplify-android/issues/1345

*Description of changes:*
- Remove invocation of error callback in the download/upload state change listener
- Add test for duplicate error callbacks 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/operation/AWSS3StorageDownloadFileOperation.java [hunk 1]
```diff
@@ -173,11 +173,7 @@ public void onStateChanged(int transferId, TransferState state) {
                     onSuccess.accept(StorageDownloadFileResult.fromFile(file));
                     return;
                 case FAILED:
-                    onError.accept(new StorageException(
-                            "Storage download operation was interrupted.",
-                            "Please verify that you have a stable internet connection."
-                    ));
-                    return;
+                    // no-op;
```
Comment (Jordan-Nelson) [inline aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/operation/AWSS3StorageDownloadFileOperation.java:176]: *Note:* `onError.accept` is already invoked in an error scenario on line 195 below

https://github.com/aws-amplify/amplify-android/blob/b7c442e92c722c2b853d2efc6c79143859a4ad62/aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/operation/AWSS3StorageDownloadFileOperation.java#L191-L200

File: testutils/src/main/java/com/amplifyframework/testutils/Await.java [hunk 2]
```diff
@@ -146,6 +146,9 @@ private Await() {}
                         resultContainer.set(result);
                         latch.countDown();
                     }, error -> {
+                        if (errorContainer.get() != null) {
+                            throw new RuntimeException("Error callback called more than once with error: " + error);
+                        }
```
Comment (raphkim) [inline testutils/src/main/java/com/amplifyframework/testutils/Await.java:151]: Great addition for catching this type of issues that were previously undetected!

File: aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/operation/AWSS3StorageUploadFileOperation.java [hunk 3]
```diff
@@ -185,11 +185,7 @@ public void onStateChanged(int transferId, TransferState state) {
                     onSuccess.accept(StorageUploadFileResult.fromKey(getRequest().getKey()));
                     return;
                 case FAILED:
-                    onError.accept(new StorageException(
-                            "Storage upload operation was interrupted.",
-                            "Please verify that you have a stable internet connection."
-                    ));
-                    return;
+                    // no-op;
```
Comment (Jordan-Nelson) [inline aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/operation/AWSS3StorageUploadFileOperation.java:188]: *Note:* As above, `onError.accept` is already invoked on line 207 below

https://github.com/aws-amplify/amplify-android/blob/b7c442e92c722c2b853d2efc6c79143859a4ad62/aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/operation/AWSS3StorageUploadFileOperation.java#L203-L212

File: testutils/src/main/java/com/amplifyframework/testutils/Await.java (hunk 2 above)
Comment (Jordan-Nelson) [reply testutils/src/main/java/com/amplifyframework/testutils/Await.java:151]: Credit to @richardmcclellan for sharing this with me 😄

**Review by raphkim — APPROVED** [review]
